asteroid/data/avspeech_dataset.py

```diff
@@ -0,0 +1,234 @@
+            A Speaker-Independent Audio-Visual Model for Speech Separation' Ephrat et. al
+            https://arxiv.org/abs/1804.03619
+    """
+    dataset_name = "AVSpeech"
+
+    def __init__(self, input_df_path: Union[str, Path], embed_dir: Union[str, Path], input_audio_size=2):
+        if isinstance(input_df_path, str):
+            input_df_path = Path(input_df_path)
+        if isinstance(embed_dir, str):
+            embed_dir = Path(embed_dir)
+
+        self.input_audio_size = input_audio_size
+        self.embed_dir = embed_dir
+        self.input_df = pd.read_csv(input_df_path.as_posix())
+        self.stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400, stride=160))
+
+    @staticmethod
+    def encode(x: np.ndarray, p=0.3, stft_encoder=None):
+        if stft_encoder is None:
+            stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400, stride=160))
+
+        x = torch.from_numpy(x).float()
+
+        # time domain to time-frequency representation
+        tf_rep = stft_encoder(x).squeeze(0) + EPS
+        # power law on complex numbers
+        tf_rep = (torch.abs(tf_rep) ** p) * torch.sign(tf_rep)
+        # (514, 298) -> (257, 298, 2)
+        tf_rep = transforms.to_torchaudio(tf_rep)
+        return tf_rep
+
+    @staticmethod
+    def decode(tf_rep: np.ndarray, p=0.3, stft_decoder=None, final_len=48000):
+        if stft_decoder is None:
+            stft_decoder = Decoder(STFTFB(n_filters=512, kernel_size=400, stride=160))
+
+        tf_rep = torch.from_numpy(tf_rep).float()
+
+        # (257, 298, 2) -> (514, 298)
+        tf_rep = transforms.from_torchaudio(tf_rep) + EPS
+        # power law on complex numbers
+        tf_rep = (torch.abs(tf_rep) ** (1 / p)) * torch.sign(tf_rep)
+        # time domain to time-frequency representation
+        x = stft_decoder(tf_rep)
+
+        length = len(x)
+        if length != final_len:
+            x = F.pad(x, [0, final_len - length])
+        return x
+
+    def __len__(self):
+        return len(self.input_df)
+
+    def __getitem__(self, idx):
+        row = self.input_df.iloc[idx, :]
+        all_signals = []
+
+        for i in range(self.input_audio_size):
+            # get audio, video path from combination dataframe
+            video_path = row.loc[f"video_{i+1}"]
+            audio_path = row.loc[f"audio_{i+1}"]
+
+            # video length is 3-10 seconds, hence, part index can take values 0-2
+            re_match = re.search(r"_part\d", audio_path)
+            video_length_idx = 0
+            if re_match:
+                video_length_idx = int(re_match.group(0)[-1])
+
+            signal = Signal(
+                video_path,
+                audio_path,
+                self.embed_dir,
+                video_start_length=video_length_idx,
+            )
+            all_signals.append(signal)
+
+        # input audio signal is the last column.
+        mixed_signal = Signal.load_audio(row.loc["mixed_audio"])
```

Comment:
And then load with librosa here. 
At least, we know we don't do anything fancy with the loading, without reading other lines of code.